spec/tree_spec.rb

```diff
@@ -1,14 +1,240 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
-  it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+describe Tree do
+  let(:tree) { Tree.new }
+
+  it 'should be an AppleTree' do
+    expect(tree.is_a? Tree).to be true
+  end
+
+  it 'should have a height of 0' do
+    expect(tree.height).to be 0  
+  end
+
+  it 'should have an age of 0' do
+    expect(tree.age).to be 0
+  end
+
+  it 'should have an empty set of apples' do
+    expect(tree.apples).to eq []
+  end
+
+  describe '#age!' do
+    
+    context 'after aging once' do
+      before(:each) { tree.age! }
+
+      it 'age is 1' do
+        expect(tree.age).to be 1
+      end
+
+      it 'height is 1' do
+        expect(tree.height).to be 1
+      end
+
+      it 'has 5 apples' do
+        expect(tree.apples.length).to be 5     
+      end
+
+    end
+    
+    context 'after aging twice' do
+      before(:each) { tree.age!; tree.age! }
+
+      it 'age is 2' do
+        expect(tree.age).to be 2
+      end
+      
+      it 'height is 2' do
+        expect(tree.height).to be 2
+      end
+
+      it 'has 15 apples' do
+        expect(tree.apples.length).to eq 15
+      end
+
+    end 
+
+    context 'after aging to 100' do
+      before(:each) do
+        tree.age = 99
+        tree.height = 99
+        tree.age!
+      end
+
+      it 'age is 100' do
+        expect(tree.age).to eq 100
+      end
+
+      it 'height is 100' do
+        expect(tree.age).to eq 100
+      end
+
+      it 'is dead' do
+        expect(tree.dead?).to be true
+      end
+
+      it 'has 500 apples' do
+        expect(tree.apples.length).to eq 500
+      end
+
+    end
+
+  end
+
+  describe '#add_apples' do
+    
+    context 'when tree is 1 year old' do
+      before(:each) do
+        tree.age = 1
+        tree.add_apples
+      end      
+      
+      it 'should add 5 apples' do
+        expect(tree.apples.length).to eq 5
+      end
+
+      it 'apples should contain apple objects' do
+        expect(tree.apples[0].is_a? Apple).to be true
+      end
+
+    end
+
+    context 'when tree is 10 years old' do
+      before :each do
+        tree.age = 10
+        tree.add_apples
+      end
+
+      it 'should add 50 apples' do
+        expect(tree.apples.length).to eq 50
+      end
+    end
+
+  end
+
+  describe '#grown_apple' do
+    
+    it 'should be red' do
+      expect(tree.grown_apple.color).to eq "red"
+    end
+
+    it 'should be atleast 5 in diameter' do
+      expect(tree.grown_apple.diameter).to be >= 5
+    end
+
+    it 'should be at most 15 in diameter' do
+      expect(tree.grown_apple.diameter).to be <= 15
+    end
```

Comment:
If the behavior is truly random, it could be a better test to just say 'the diameter exists' or 'the diameter is a number', since you'll never really be able to tell what value it's supposed to be.